astropy/units/quantity_helper/function_helpers.py

```diff
@@ -695,16 +748,14 @@ def array_impl(object, *, dtype, copy, order, subok, ndmin, ndmax=0):
 
 
 if NUMPY_LT_2_0:
-    asarray_impl_1_helps = {np.asarray, np.asanyarray}
-    asarray_impl_2_helps = {}
+    asarray_impl_1_helps = {"asarray", "asanyarray"}
+    asarray_impl_2_helps = set()
```

Comment:
nice catch, this is actually a bug. Possibly harmless, and out of scope here, but a bug none the less. If you would be so kind as to open another PR with just this (and similar) change(s), I'd merge it instantly.